test/fixtures/engine/engine-dependencies/root.hcl

```diff
@@ -0,0 +1,10 @@
+remote_state {
+  backend = "local"
+  generate = {
+    path = "backend.gen.tf"
+    if_exists = "overwrite_terragrunt"
+  }
+  config = {
+    path = "${get_terragrunt_dir()}/terraform.tfstate"
```

Comment:
NIT: always bias towards OpenTofu. Don't change this now, we can edit after this PR.